pkg/gofr/gofr.go

```diff
@@ -331,10 +338,15 @@ func (a *App) PATCH(pattern string, handler Handler) {
 func (a *App) add(method, pattern string, h Handler) {
 	a.httpRegistered = true
 
+	reqTimeout, err := strconv.Atoi(a.Config.Get("REQUEST_TIMEOUT"))
+	if (err != nil && a.Config.Get("REQUEST_TIMEOUT") != "") || reqTimeout < 0 {
+		reqTimeout = 0
+	}
```

Comment:
I agree with @srijan-27 

The code is duplicated

Can't we move the log about the error here?

Reply:
Moving the logger here will have this impact as told earlier:
https://github.com/gofr-dev/gofr/pull/831#discussion_r1708752604

Reply:
yes, `add` is called for all routes so the log can't be moved here.. same is the case for config (why do we need to validate config for every route) as it is defined throughout the application.

Reply:
indeed. Thanks for reminding me.

Can't add a requestTimeout into the App struct, define it when reading the conf, and simply use it here?

Reading the env var, validating it, on each call to `add` seems a problem to me, even more when the code is duplicated, so it means code may divert, and could be a pain to maintain

Reply:
IMO, I don't think so putting `request-timeout` field in app struct is a better option. The GoFr app struct hold the servers, conatiner, subscriptionManager etc. , putting an config field there may pollute the struct

Reply:
OK, I can agree that. 

One last suggestion: what about having a method to get the value from the env variable then?

Reply:
We already have method to get value from Config:

```go

func (*EnvLoader) Get(key string) string {
	return os.Getenv(key)
}

func (*EnvLoader) GetOrDefault(key, defaultValue string) string {
	if val := os.Getenv(key); val != "" {
		return val
	}

	return defaultValue
}
```

Are you suggesting we should have a method on app to Get env variables ?

Reply:
Yes, I'm thinking about something like this

```go
func (*App) getRequestTimeout (vale int, valid bool) {
	
      requestTimeout := a.Config.Get("REQUEST_TIMEOUT") 
      if requestTimeout == "" {
            return 0, true
      }
      timeoutVal, err := strconv.Atoi(requestTimeout) 
      if err != nil || timeoutVal < 0 { 
             return 0, false // can be checked to log an error
      }

      return timeoutVal, true
}
```

Reply:
@srijan-27 @vipul-rawat what do you guys think? Should we make a method on app for this use-case?

Reply:
I guess creating a method also will not rectify our main concern.. (reading config multiple times) as we would anyways be calling this method in `add`.
Not a good thing to do, but should we create a local variable and update the value for this while reading the config in `httpServerSetup`, and using this variable in `add`?

Reply:
That was my suggestion 😅😁

Reply:
if we will create a variable in `gofr.go` file then it will be a global variable. even if you keep it unexported but still it will be global on scope. And we should not create global variables untill very necessary. 
@srijan-27 do you agree?

Reply:
couldn't think of any viable solution.. lets add a TODO for this.. and take it later, let's close this PR first since it is pending for long time.